crates/core/src/workflows/application.rs

```diff
@@ -66,8 +68,24 @@ pub async fn initialize(
     initialize_network(&instance_context).await?;
 
     // 4. (first initialization only) Call the init callbacks in the zomes
+    //     and publish the headers of the initial chain entries
     if first_initialization {
         call_init(dna, &instance_context).await?;
+
+        log_debug!(
+            context,
+            "dna/initialize: publishing headers of initial chain entries...",
+        );
+
+        for header in context.state().unwrap().agent().iter_chain() {
```

Comment:
@maackle, can you confirm that this change here doesn't require us updating the consistency model?

Reply:
As long as everyone holds these header entries like any other entry (via the `Hold` action reducer), there should be no change needed. In fact I think we could simplify the model by taking out the InitNetwork event which we added to handle the special case of the initial entries (not that we actually should do that)

Reply:
I don't think we can rely on this explicit publishing of anything during initialization. We rely on the authoring and gossip lists being correct. If tests fail without this explicit publish it means that the lists are not correct (missing these addresses). Making the tests work with this is just faking the tests and not fixing the problem for the case where we have complex network topologies and nodes coming and going, which we are not completely testing right now. So I think I will change my approval to requesting this to be removed and fixed by looking to why the authoring/gossip lists are not enough..

Reply:
@lucksus also had a concern about publishing here only fixing the problem in one case and masking validation problems.